### PR TITLE
Use TestEnv

### PR DESCRIPTION
**Pull request by mmiller-max**
I've just merged in main and this seems to be passing, picking up my work from over two years ago! Unfortunately I don't remember why this wasn't working before or why I wasn't happy with it, possibly because it failed for V1 (but then I dropped tests for V1). Would be grateful for @oxinabox and/or @omus to take a look at this 🙏 
Hopefully will fix a few issues (e.g. #121 #122 )

edit: Just added back in V1 tests so will see what happens
edit 2: They fail, and they fail on master too (at least locally for me). At this point I think the best way forward is to remove 1.0 unless anyone has any objections?

### CODE REVIEW

File: src/runner.jl
Comment (oxinabox) [inline src/runner.jl:259]: ideally we wouldn't use the TestEnv internals here.
Because we would like to contain all the evil of accessing interals of Pkg into TestEnv so TestReports can be safe from it

**Review by oxinabox — APPROVED** [review]
This is clearly a solid improvement

File: src/runner.jl
Comment (oxinabox) [reply src/runner.jl:259]: but i don't think should block on this

Comment (mmiller-max) [reply src/runner.jl:259]: Added a TODO and will make an issue after merge

**Pull request merged**
